src/app/routes/cpsAsset/getInitialData/convertToOptimoBlocks/blocks/include/includeClassifier.test.js

```diff
@@ -0,0 +1,157 @@
+    const expected = { includeType: 'idt2', classification: 'idt2-amp' };
+
+    const actual = includeClassifier({
+      href: idt2Include,
+      pathname: ampPathname,
+    });
+
+    expect(actual).toEqual(expected);
+  });
+
+  it('should classify VJ include supporting AMP on Amp platform', () => {
+    const expected = {
+      includeType: 'vj',
+      classification: 'vj-include-supports-amp',
+    };
+
+    const actual = includeClassifier({
+      href: vjIncludeSupportingAmp,
+      pathname: ampPathname,
+    });
+
+    expect(actual).toEqual(expected);
+  });
+
+  it('should classify VJ include not supporting AMP on Amp platform', () => {
+    const expected = {
+      includeType: 'vj',
+      classification: 'vj-include-not-supporting-amp',
+    };
+
+    const actual = includeClassifier({
+      href: vjIncludeNotSupportingAmp,
+      pathname: ampPathname,
+    });
+
+    expect(actual).toEqual(expected);
+  });
+
+  it('should classify VJ include supporting AMP on canonical platform', () => {
+    const expected = {
+      includeType: 'vj',
+      classification: 'vj-include-canonical',
+    };
+
+    const actual = includeClassifier({
+      href: vjIncludeSupportingAmp,
+      pathname: canonicalPathname,
+    });
+
+    expect(actual).toEqual(expected);
+  });
+
+  it('should classify VJ include not supporting AMP on canonical platform', () => {
+    const expected = {
+      includeType: 'vj',
+      classification: 'vj-include-canonical',
+    };
+
+    const actual = includeClassifier({
+      href: vjIncludeNotSupportingAmp,
+      pathname: canonicalPathname,
+    });
+
+    expect(actual).toEqual(expected);
+  });
+
+  it('should classify a non-supported include as not supported on canonical', () => {
+    const expected = { includeType: null, classification: 'not-supported' };
+
+    const actual = includeClassifier({
+      href: 'idt3/blah',
+      pathname: canonicalPathname,
+    });
+
+    expect(actual).toEqual(expected);
+  });
+
+  it('should classify a non-supported include as not supported on amp', () => {
+    const expected = { includeType: null, classification: 'not-supported' };
+
+    const actual = includeClassifier({
+      href: 'idt3/blah',
+      pathname: ampPathname,
+    });
+
+    expect(actual).toEqual(expected);
+  });
+
+  it.each`
+    includeType | classification            | href
+    ${`idt1`}   | ${`idt1-canonical`}       | ${`indepthtoolkit/quizzes/123-456`}
+    ${`idt2`}   | ${`idt2-canonical`}       | ${`idt2/111-222-333-444-555`}
+    ${`vj`}     | ${`vj-include-canonical`} | ${`include/newsspec/21841-green-diet/gahuza/app?responsive=true&newsapps=true&app-image=https://news.files.bbci.co.uk/vj/live/idt-images/image-slider-asdf/app_launcher_ws_640_7ania.png&app-clickable=true&amp-clickable=true&amp-image-height=360&amp-image-width=640&amp-image=https://news.files.bbci.co.uk/vj/live/idt-images/image-slider-asdf/app_launcher_ws_640_7ania.png`}
+  `(
```

Comment:
I've used a for loop in the past, this looks a much a neater approach 👍

Reply:
May want to add tests for the leading slash for amp too? Be careful not to overcomplicate this test though, better to duplicate test for Amp than have more and more rows in your table above.

Reply:
This method looks great!